packages/material-ui/src/Divider/Divider.js

```diff
@@ -1,181 +1,252 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import { useThemeVariants } from '@material-ui/styles';
-import withStyles from '../styles/withStyles';
+import { deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import experimentalStyled from '../styles/experimentalStyled';
 import { alpha } from '../styles/colorManipulator';
+import dividerClasses, { getDividerUtilityClass } from './dividerClasses';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...styles[styleProps.variant],
+    ...(styleProps.absolute && styles.absolute),
+    ...(styleProps.flexItem && styles.flexItem),
+    ...(styleProps.light && styles.light),
+    ...(styleProps.orientation === 'vertical' && styles.vertical),
+    ...(styleProps.children && styles.withChildren),
+    ...(styleProps.children &&
+      styleProps.orientation === 'vertical' &&
+      styles.withChildrenVertical),
+    ...(styleProps.textAlign === 'right' &&
+      styleProps.orientation !== 'vertical' &&
+      styles.textAlignRight),
+    ...(styleProps.textAlign === 'left' &&
+      styleProps.orientation !== 'vertical' &&
+      styles.textAlignLeft),
+    [`&.${dividerClasses.wrapper}`]: styles.wrapper,
```

Comment:
```suggestion
    [`& .${dividerClasses.wrapper}`]: styles.wrapper,
```
This is a child selector